### PR TITLE
Add Wifi support

### PR DESCRIPTION
**Pull request by psppalash**
This feature allows maestro to connect to wifi by using wpa-connect package.
Wifi SSID, Wifi Passsword fields are added to network config.
The wifi config can be changed through maestro-shell and devicedb.

### CODE REVIEW

**Comment by psppalash** [comment]
wpa-connect and 4 other thrid party packages were added to the vendor directory.
Additionally, wpa-connect package (MIT license) had to be modified to get certain functionality working.

**Comment by psppalash** [comment]
Wifi support was tested on a DELL gateway by changing network config through maestro-shell and switching between 2 wifi networks

**Comment by psppalash** [comment]
@costanic I think I will need your help in using godep (https://github.com/tools/godep). I tried using it but I am facing a lot of issues.

**Comment by costanic** [comment]
@psppalash ok, since Godep is giving you problems let's figure out a way for you to move forward in the short term in a way that makes it easier to switch to go-modules.

**Comment by psppalash** [comment]
> @psppalash ok, since Godep is giving you problems let's figure out a way for you to move forward in the short term in a way that makes it easier to switch to go-modules.

Sure. I can schedule a meeting so that we can discuss how to work towards the switch to go-modules.

**Comment by psppalash** [comment]
@costanic The URL's and SHA for the packages that are added in this PR.
https://github.com/creack/pty           2a38352e8b4d7ab6c336eef107e42a55e72e7fbc
https://github.com/godbus/dbus          efee8394fa9a426004eb24eed1dee80f5efd90af
https://github.com/kballard/go-shellquote   95032a82bc518f77982ea72343cc1ade730072f0
https://github.com/kr/pty                    24b51915630e1d9d3c6adf15a3a96796bbc3cd43
https://github.com/ThomasRooney/gexpect      5482f03509440585d13d8f648989e05903001842

**Comment by psppalash** [comment]
Pr for https://github.com/armPelionEdge/wpa-connect/pull/1

**Comment by psppalash** [comment]
@costanic I changed the commits. Please have a look. Please do not merge this yet. I want to test it once again on hardware.
BTW, I think maestro-shell should return an error if there was an error while setting up an interface. I don't think maestro-shell does that right now. Do you think we should add this feature?

**Comment by ryannowarm** [comment]
@psppalash, seems reasonable.  Add a story in the backlog for that.

File: networking/manager.go [hunk 1]
```diff
@@ -655,6 +655,12 @@ func validateIfConfig(ifconf *maestroSpecs.NetIfConfigPayload) (ok bool, problem
 		ok = false
 		problem = "Interface is missing IfName / if_name field"
 	}
+	if ifconf.Type == "wifi" {
```
Comment (costanic) [inline networking/manager.go:658]: Now that we support multiple types, should this function check to make sure that ifconf.Type exists is one of our supported types?

Comment (psppalash) [reply networking/manager.go:658]: Yes, that's a good idea. But the problem is in order to configure ethernet interfaces, maestro documentation and commands do not require ifconf.Type to be set (i.e it is an empty string). Do we just return an error if ifconf.Type is not an empty string or our supported types?

Comment (costanic) [reply networking/manager.go:655]: Defaulting to ethernet if the type is empty string seems fine.  Is there a central place where defaults are provided?

**Comment by psppalash** [comment]
Tested on hardware. @costanic @mray190 This PR is ready to be merged. Also, do you know why the CI check is failing?

File: networking/manager.go [hunk 2]
```diff
@@ -1036,7 +1036,7 @@ func (this *networkManagerInstance) setupInterfaces() (err error) {
 			task.Src = "SetupExistingInterfaces"
 			task.Op = op
 
-			err = this.SubmitTask(task)
```
Comment (costanic) [inline networking/manager.go]: Can you add a git-commit message as to why this revert was needed?  Or better yet, amend the commit that introduced this change if that commit is also contained within this PR?

Comment (psppalash) [reply networking/manager.go]: Done.

**Pull request merged**